Warehouse FAQ revamp and addition (#590)
rudderlabs/rudderstack-docs: docs/data-warehouse-integrations/warehouse-faqs.mdx

# slug: "/docs/data-warehouse-integrations/warehouse-faqs"
title: "FAQ"
description: >-
  Commonly-asked questions for the RudderStack data warehouse integrations.
---

# FAQ

### When does Rudderstack load data into the data warehouse(s)?

> Rudderstack lets you choose when to sync the latest data into the warehouse. The default time is 30 minutes, however you can choose to extend this value upto 24 hours. You can also configure the time at which the data needs to be loaded.

Reviewer: ```suggestion
RudderStack lets you choose how frequently it should sync the data into the data warehouse. The default time is 30 minutes. However, you can choose to extend this value up to 24 hours. You can also configure the time of the day when RudderStack loads the data.
```

### Is there a way to force load my data into the warehouses?

Yes, there is - you can configure the below values in your `config.yaml` file:

> ```

Reviewer: ```suggestion
```yaml
```

warehouseSyncFreqIgnore = true #if set true this will ignore syncFrequency and syncStartingAt values which are configured in UI. By default this is false
uploadFreqInS=1800 #This field lets you control syncPeriod if above field set to true 
```

### Where can I view the status of my data load?

> The warehouse upload status can be viewed in **Live Events** section of the destination in the RudderStack dashboard.

Reviewer: ```suggestion
You can view the warehouse upload status in the **Live Events** section of the destination in the RudderStack dashboard.
```

### Can I change the namespace (schema name) of my data warehouse in RudderStack?

Yes, you can. Although the default namespace will be the source name with some modifications to it, RudderStack also provides an option in the dashboard to explicitly set the namespace of the dataset in the warehouse.

<div class="infoBlock">

Refer to the warehouse-specific destination settings to configure a namespace in the RudderStack dashboard.
</div>

### How can I achieve identity resolution with multiple sources connected to one warehouse destination?

> The default namespace will be the source name with some modifications. This will cause issues while creating mappings for Identity Resolution across different sources, leading to multiple `RUDDER_ID` being created for the same user.

Reviewer: ```suggestion
The default namespace will be the source name with some modifications. This will cause issues while creating mappings for identity resolution across different sources, leading to multiple `RUDDER_ID` created for the same user.
```

> To avoid this, simply match the namespace for each source.

Reviewer: ```suggestion
To avoid this, you can set the same namespace for each source.
```

<img src="../assets/dw-integrations/namespace.png" />

> This will ensure that the mappings are on the consistent namespace and only one `RUDDER_ID` is created for each user no matter how many sources are connected to your warehouse destination.

Reviewer: ```suggestion
This will ensure that the mappings are on the consistent namespace and only one `RUDDER_ID` is created for each user - irrespective of the number of sources connected to your warehouse destination.
```

### How can I delete the staging files in my buckets?

> RudderStack loads all the events configured with your warehouse into staging buckets. You can set a retention policy to delete these files from the staging bucket after a certain amount of time. If no retention policy is set, the files will keep on accumulating.

Reviewer: ```suggestion
RudderStack loads all the events configured with your warehouse into staging buckets. You can set a retention policy to delete these files from the staging bucket after a certain time. If you do not set any retention policy, the files will keep on accumulating.
```

> Make sure that the duration of retention policy is longer than the warehouse sync frequency duration. The reason being that RudderStack can re-run event uploads for a longer time, if required. It is recommeded to set the retention policy duration to **30 days/1 month**. 

Reviewer: ```suggestion
Make sure that the retention policy duration is longer than the warehouse sync frequency duration so that RudderStack can re-run event uploads for a longer time, if required. It is recommended to set the retention policy duration to **30 days/1 month**. 
```

### How can I speed-up my warehouse uploads?

> You can speed-up your warehouse uploads by using the following parameters:

Reviewer: ```suggestion
You can speed up your warehouse uploads by configuring the following parameters in the `config.yaml` file of your RudderStack setup:
```

    <td><code class="inline-code">960</code></td>
  </tr>
  <tr>
    <td style="width:300px"><code class="inline-code">RSERVER_WAREHOUSE_NO_OF_WORKERS</code></td>
    <td>Number of concurrent uploads to a warehouse. For example, 8 uploads to different schemas in Redshift can be done simultaneously.</td>
    <td>-</td>
    <td><code class="inline-code">8</code></td>
  </tr>
   <tr>
    <td style="width:300px"><code class="inline-code">warehouse_slave.replicaCount</code></td>
    <td>Number of warehouse slaves (responsible for load file generation) when the warehouse is run as separate service in Kubernetes.</td>
    <td>Increase this for speeding up the load file generation step of the upload.</td>
    <td><code class="inline-code">2</code></td>
  </tr>
   <tr>
    <td style="width:300px"><code class="inline-code">RSERVER_WAREHOUSE_NO_OF_SLAVE_WORKER_ROUTINES</code></td>
    <td>Number of go-routines creating load files in a warehouse slave process.</td>
    <td>Increase or decrease this as per memory allocated to the warehouse slave pod.</td>
    <td><code class="inline-code">4</code></td>
  </tr>
</table>

### How is the Namespace field populated for a Warehouse destination?

> The Namespace field depends on the following attributes:

Reviewer: ```suggestion
RudderStack sets the **Namespace** field depending on the following conditions:
```

> - If the Namespace and Database fields are present:

Reviewer: ```suggestion
- If the **Namespace** and **Database** fields are present in your warehouse configuration:
```

>   - For Clickhouse destination, the Database field is populated as Namespace.

Reviewer: ```suggestion
  - For ClickHouse destination, the **Database** field is set as the namespace.
```

>   - For all the other destinations, the Namespace field is populated as it is.

Reviewer: ```suggestion
  - For all the other destinations, the **Namespace** field is populated as it is.
```

- If `RSERVER_WAREHOUSE_(DEST_TYPE)_CUSTOM_DATA_SET_PREFIX` parameter is present, the Namespace is populated as `(DEST_TYPE)_(CUSTOM_DATA_SET_PREFIX)_(SOURCE_NAME)`. For example:

```
Parameter value:
RSERVER_WAREHOUSE_RS_CUSTOM_DATA_SET_PREFIX="<random_value>"
RSERVER_WAREHOUSE_SNOWFLAKE_CUSTOM_DATA_SET_PREFIX="<random_value>"

Namespace:
RS_<random_value>_<source_name>
SNOWFLAKE_<random_value>_<source_name>
```

- If none of the above are specified, the source name is populated as Namespace for the first sync and picked up from cache for the subsequent syncs. In case the source name is changed, the Namespace still remains the same as it is picked from cache. In order to update the Namespace as the new source name, you need to configure a new source.

### How does Rudderstack configure table names?

> The table names are picked up from the event names and are modified to follow the snake case convention (for example, source name -> source_name). 

Reviewer: ```suggestion
RudderStack sets the table names by picking up the event names and modifying them to follow the snake case convention (for example, `source name` -> `source_name`). 
```

> For the following sample snippet, the table name is set as **product_purchased**.

Reviewer: ```suggestion
For the following sample snippet, RudderStack sets the table name as **product_purchased**.
```

```json
{
  "userId": "user123",
  "event": "Product Purchased",
  "properties": {
    "name": "Rubik's Cube",
    "revenue": 4.99
  },
  "context": {
    "ip": "14.5.67.21"
  },
  "timestamp": "2020-02-02T00:23:09.544Z"
}
```

The table name truncation logic is as follows:

- For **PostgreSQL**, the table name is truncated after 63 characters.
- For **Data Lake** destinations (**S3 Data Lake, GCS Data Lake, Azure Data Lake**), there is no set limit.
- For **other** destinations, the table name is truncated after 127 characters.

### How does Rudderstack configure the column names? 

### How does RudderStack normalize arrays and complex event properties in the warehouse schema? 

> RudderStack flattens out the event properties and converts them to snake case (for example, source name -> source_name). Further, each of these event properties act as a column for the warehouse.

Reviewer: ```suggestion
RudderStack flattens the event properties and converts them into snake case (for example, `source name` -> `source_name`). Further, each of these event properties acts as a warehouse column.
```

1. The below example shows how the normal properties are normalized:

```javascript
{
  product: {
    name: "iPhone",
    version: 11
  }
}
```

Normalized properties (Column names)

```javascript
product_name : "iPhone"
product_version : 11
```

2. The below example shows how the array properties are normalized:

```javascript
{
  products: [
    {
      name: "iPhone",
      version: 11
    },
    {
      name: "Android",
      version: 16
    }
  ]
}
```

Normalized property (Column name)

```javascript
products: "[{ \"name\": \"iPhone\", \"version\": 11  },  { \"name\": \"Android\", \"version\": 16  }]"
```

3. The below example shows how the complex properties are normalized:

```javascript
{
  products: {
    iPhone: {
      version: 11
    },
    Android: {
      version: 16
    }
  }
}
```

Normalized properties (Column names)

```javascript

> products_i_phone_version : 11

Reviewer: ```suggestion
products_iphone_version : 11
```
